addons/stock/report/report_stock_quantity.py

```diff
@@ -38,18 +38,18 @@ def init(self):
         m.id,
         m.product_id,
         CASE
-            WHEN (whs.id IS NOT NULL AND whd.id IS NULL) OR ls.usage = 'transit' THEN 'out'
-            WHEN (whs.id IS NULL AND whd.id IS NOT NULL) OR ld.usage = 'transit' THEN 'in'
+            WHEN whs.id IS NOT NULL AND (whd.id IS NULL OR ld.usage = 'transit') THEN 'out'
```

Comment:
testing the transit usage was needed when we tested also that a location was internal. Now the test is done on the warehouse presence or not It seems to be useless now. Transit location should not have any warehouse above them.

I guess you can even simplify the condition like this
```sql
WHEN whs.id IS NOT NULL AND whd.id IS NULL THEN 'out'
WHEN whs.id NOT NULL AND whd.id IS NOT NULL THEN 'in
```

Reply:
@Whenrow I have modified.